src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -493,6 +493,91 @@ trait GoldTransforms extends SparkSessionWrapper {
 
   }
 
+
+  protected def buildNotebookCommands(
+                                   notebook: PipelineTable,
+                                   clsfIncrementalDF : DataFrame,
+                                 )(auditIncrementalDF: DataFrame): DataFrame = {
+
+
+    val notebookLookupTSDF = notebook.asDF
+      .select("organization_id", "notebook_id", "notebook_path", "notebook_name", "unixTimeMS","date")
+      .withColumnRenamed("notebook_id", "notebookId")
+      .filter('notebookId.isNotNull)
+      .distinct
+      .toTSDF("unixTimeMS", "organization_id","date","notebookId")
+
+    val clsfLookupTSDF = clsfIncrementalDF
+      .select(
+        "organization_id", "state_start_date", "unixTimeMS_state_start", "cluster_id", "cluster_name",
+        "custom_tags", "node_type_id", "current_num_workers", "uptime_in_state_H", "total_cost")
+      .distinct
+      .withColumnRenamed("unixTimeMS_state_start", "unixTimeMS")
+      .withColumnRenamed("cluster_id", "clusterId")
+      .withColumnRenamed("current_num_workers", "node_count")
+      .withColumn("totalCostPMS", col("total_cost") / col("uptime_in_state_H"))
+      .toTSDF("unixTimeMS", "organization_id", "clusterId")
+
+    val colNames: Array[Column] = Array(
+      'organization_id,
+      'clusterId.alias("cluster_id"),
+      'notebookId.alias("notebook_id"),
+      'workspace_name,
+      'date,
+      'timestamp,
+      'notebook_path,
+      'notebook_name,
+      'commandId.alias("command_id"),
+      'commandText.alias("command_text"),
+      'executionTime.alias("execution_time_s"),
+      'sourceIpAddress.alias("source_ip_address"),
+      'userIdentity.alias("user_identity"),
+      'estimated_dbu_cost,
+      'status,
+      'cluster_name,
+      'custom_tags,
+      'node_type_id,
+      'node_count,
+      'response,
+      'userAgent.alias("user_agent"),
+      'unixTimeMS
+    )
+
+    val notebookCodeAndMetaDF = auditIncrementalDF
+      .filter(col("serviceName") === "notebook" && col("actionName") === "runCommand")
+      .selectExpr("*", "requestParams.*").drop("requestParams")
+      .withColumnRenamed("timestamp", "unixTimeMS")
+      .withColumn("timestamp", from_unixtime(col("unixTimeMS") / 1000.0).cast("timestamp"))
+      .withColumn("executionTime",col("executionTime").cast("double"))
+      .filter('notebookId.isNotNull)
+      .toTSDF("unixTimeMS", "organization_id", "date","notebookId")
+      .lookupWhen(
+        notebookLookupTSDF,
+        maxLookAhead = 100
+      )
+      .df
+      .toTSDF("unixTimeMS", "organization_id", "clusterId")
+      .lookupWhen(
+        clsfLookupTSDF,
+        tsPartitionVal = 8
+      )
+      .df
+      .withColumn("estimated_dbu_cost", col("executionTime") * col("totalCostPMS"))
```

Comment:
looks like there's an issue here

something that runs on 8 nodes for 1275 seconds CANNOT cost $30K. ;) Please add some logic to your proof to do a sanity check on estimated costs.

![image](https://github.com/databrickslabs/overwatch/assets/10840635/e4c9e119-0109-44b7-bdcf-74c63ab01b22)